### PR TITLE
Highlighting panel and 'Data view settings' button to show where error occurred

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@tonytlwu @squallstar 
## Issue
https://github.com/Fliplet/fliplet-studio/issues/4877

## Description
Added error highlighting for panel and 'Data view settings' button

## Screenshots/screencasts
issue-4877

## Backward compatibility

This change is fully backward compatible.

### CODE REVIEW

File: js/interface-lists.js [hunk 1]
```diff
@@ -208,6 +208,19 @@ function validate(value) {
...
 
+function showOrHideError (showError, element) {
+  if (showError) {
+    var $element = $(element);
+    $element.addClass('has-error');
+    $element.parents('.form-group').addClass('has-error');
+    $element.parents('.panel').addClass('panel-danger').removeClass('panel-default');
+    return;
+  }
+    $('.has-error').removeClass('has-error');
```
Comment (tonytlwu) [inline js/interface-lists.js]: Unindent this section, and add a linebreak after the `if` clause

File: js/interface-lists.js [hunk 2]
```diff
@@ -208,6 +208,19 @@ function validate(value) {
   return false;
 }
 
+function showOrHideError (showError, element) {
```
Comment (tonytlwu) [inline js/interface-lists.js]:
```suggestion
function toggleError (showError, element) {
```

File: js/interface-lists.js [hunk 3]
```diff
@@ -452,8 +481,7 @@ function attahObservers() {
             }, 4000);
             return;
           } else {
-            $('.has-error').removeClass('has-error');
-            $('.component-error').addClass('hidden');
+            showOrHideError(false);
```
Comment (tonytlwu) [inline js/interface-lists.js]: I notice a `field` isn't provided here, but there's nothing in the function that deals with what happens if the second parameter isn't provided.

File: js/interface-lists.js [hunk 4]
```diff
@@ -462,7 +490,7 @@ function attahObservers() {
           var values = [];
 
           values.push({
-            value: widgetData.pollColumn,
+            value: widgetData.surveyColumn,
```
Comment (tonytlwu) [inline js/interface-lists.js]: Why is this changed?

Comment (YaroslavOvdii) [reply js/interface-lists.js]: This is needed to avoid an error like this.
can-save-without-a-field

As you may see we able to save surveys without selecting any survey field. Because we were checking only the poll field.

File: js/interface-lists.js (hunk 3 above)
Comment (YaroslavOvdii) [reply js/interface-lists.js]: We deal with this by if clause which fires when we provide a `true` as the first parameter and exit from the function by return in if clause. In another situation, we remove error classes by referring to them by classes.

File: js/interface-lists.js (hunk 2 above)
Comment (YaroslavOvdii) [reply js/interface-lists.js]: @tonytlwu 
Renamed function

File: js/interface-lists.js (hunk 1 above)
Comment (YaroslavOvdii) [reply js/interface-lists.js]: @tonytlwu 
Added a linebreak after `if` clause

Comment (tonytlwu) [reply js/interface-lists.js]: Also unindent this section, like below

```suggestion
  $('.has-error').removeClass('has-error');
```

File: js/interface-lists.js [hunk 5]
```diff
@@ -421,6 +435,23 @@ function attahObservers() {
             $('.component-error').addClass('hidden');
           }
         }
+        var imageFolderSelected = validateImageFoldersSelection();
+        if (imageFolderSelected) {
```
Comment (tonytlwu) [inline js/interface-lists.js]: Line break before `if` clause

Comment (YaroslavOvdii) [reply js/interface-lists.js]: @tonytlwu 
Fixed

File: js/interface-lists.js (hunk 1 above)
Comment (YaroslavOvdii) [reply js/interface-lists.js]: @tonytlwu 
Fixed
